pentaho/src/main/java/pt/webdetails/cda/dataaccess/PentahoDataAccessUtils.java

```diff
@@ -115,10 +126,57 @@ public KettleTransformationProducer createKettleTransformationProducer( TransFro
     } else {
       relPath = ktrPath;
     }
-
-    return new CdaPentahoKettleTransFromFileProducer( "",
+    Repository repo = null;
+    try {
+      repo = connectToRepository();
+    } catch ( KettleException | ActionExecutionException e ) {
+      logger.warn( "Failed to connect to repository. " + e.getMessage() );
+    }
+    CdaPentahoKettleTransFromFileProducer cdaPentahoKettleTransFromFileProducer
+        = new CdaPentahoKettleTransFromFileProducer( "",
       relPath, query, null, null, connectionInfo.getDefinedArgumentNames(),
       connectionInfo.getDefinedVariableNames() );
+    cdaPentahoKettleTransFromFileProducer.setRepository( repo );
+    return cdaPentahoKettleTransFromFileProducer;
+  }
+
+  protected Repository connectToRepository() throws KettleException, ActionExecutionException {
+    RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+
+    // only load a default enterprise repository. If this option is set, then you cannot load
+    // transformations or jobs from anywhere but the local server.
+    String repositoriesXml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><repositories>" //$NON-NLS-1$
```

Comment:
Are we sure this definition is sufficient? Aren't we assuming a lot about the server repository we're using. 
We know for a fact that we're running in the context of a Pentaho Server - can't we leverage that to get a reference to the repository? Instead of a hardcoded repository xml?